### DISCUSSION TITLE
Support for AOT

### DISCUSSION THREAD

**Discussion by simboyd** [dotnet/WatsonWebserver#121] [Q&A] [answered]
Hello!
Can Watson be run as an AOT app? They just introduced this feature for ASP.NET in .NET 8. I guess a more specific question would be - does Watson use reflection, or depend on anything that uses reflection?

**Comment by jchristn** [comment, marked as answer]
Hi @simboyd !  Currently it does not support AOT, because it uses reflection for identifying routes defined by attributes.  However I plan to remove this capability for two reasons, 1) to support AOT, and 2) because it has proved to be more problematic than helpful.  It will likely be in another few weeks.  Cheers!

**Reply by jchristn** [reply]
Hi @simboyd I just released v6.0 which *should* support AOT.  Please give it a try and if it doesn't, let me know!

**Reply by simboyd** [reply]
Hi @jchristn! That's really cool, I'll keep that in mind for the future. Fortunately I didn't have to wait because I found a solution with HttpListener for my particular use case.
